server/mlabns/tests/test_tool_fetcher.py

```diff
@@ -0,0 +1,260 @@
+        tool.latitude = latitude
+        tool.longitude = longitude
+        tool.country = country
+        tool.put()
+
+    def insertSite(self, site_id):
+        site = model.Site(parent=self.db_root.key())
+        site.site_id = site_id
+        # Metro is always a 2-tuple where the first is the metro (first three
+        # letters of the site ID) and the second is the site ID itself.
+        site.metro = [site_id[:3], site_id]
+        site.put()
+
+    def initToolIdSiteGroup(self):
+        self.insertSliverTool(
+            tool_id='mock_tool_a', site_id='abc01', country='CountryA',
+            status_ipv4=message.STATUS_ONLINE,
+            status_ipv6=message.STATUS_ONLINE)
+        self.insertSliverTool(
+            tool_id='mock_tool_b', site_id='abc01', country='CountryA',
+            status_ipv4=message.STATUS_ONLINE,
+            status_ipv6=message.STATUS_ONLINE)
+        self.insertSliverTool(
+            tool_id='mock_tool_c', site_id='abc01', country='CountryA',
+            status_ipv4=message.STATUS_ONLINE,
+            status_ipv6=message.STATUS_ONLINE)
+        self.insertSliverTool(
+            tool_id='mock_tool_a', site_id='abc02', country='CountryA',
+            status_ipv4=message.STATUS_ONLINE,
+            status_ipv6=message.STATUS_OFFLINE)
+        self.insertSliverTool(
+            tool_id='mock_tool_b', site_id='abc02', country='CountryA',
+            status_ipv4=message.STATUS_OFFLINE,
+            status_ipv6=message.STATUS_OFFLINE)
+        self.insertSliverTool(
+            tool_id='mock_tool_a', site_id='abc03', country='CountryA',
+            status_ipv4=message.STATUS_OFFLINE,
+            status_ipv6=message.STATUS_OFFLINE)
+        self.insertSliverTool(
+            tool_id='mock_tool_c', site_id='abc03', country='CountryA',
+            status_ipv4=message.STATUS_OFFLINE,
+            status_ipv6=message.STATUS_OFFLINE)
+
+    def testFetchToolA(self):
+        self.initToolIdSiteGroup()
+        tool_properties = tool_fetcher.ToolProperties(tool_id='mock_tool_a')
+        self.verifyPropertiesReturnExpectedSiteIds(
+            ('abc01', 'abc02', 'abc03'), tool_properties)
+
+    def testFetchToolB(self):
+        self.initToolIdSiteGroup()
+        tool_properties = tool_fetcher.ToolProperties(tool_id='mock_tool_b')
+        self.verifyPropertiesReturnExpectedSiteIds(
+            ('abc01', 'abc02'), tool_properties)
+
+    def testFetchToolC(self):
+        self.initToolIdSiteGroup()
+        tool_properties = tool_fetcher.ToolProperties(tool_id='mock_tool_c')
+        self.verifyPropertiesReturnExpectedSiteIds(
+            ('abc01', 'abc03'), tool_properties)
+
+    def testFetchNonExistentTool(self):
+        self.initToolIdSiteGroup()
+        tool_properties = tool_fetcher.ToolProperties(tool_id='no_exist_tool')
+        self.verifyPropertiesReturnExpectedSiteIds([], tool_properties)
+
+    def initStatusSiteGroup(self):
+        self.insertSliverTool(
```

Comment:
On real mlab-ns, this data is populated by querying Nagios and building up the information in the AppEngine Datastore.